ladybug/epw.py

```diff
@@ -185,18 +186,17 @@ def _import_data(self, import_location_only=False):
                 # since I'm using this timestamp as the key and will be using it for
                 # sorting. I'm setting it up to 2015 - the real year will be collected
                 # under modelYear
-                timestamp_half_hour = DateTime(month, day, hour - 1)
-                timestamp_hour = DateTime(month, day, hour - 1)
+                timestamp_middle_hour = DateTime(month, day, hour - 1)
 
-                insert_hour = False
                 if month == 12 and day == 31 and hour == 24:
```

Comment:
What if instead of checking this here on the initial input we just adjust the hours all together at the end and only for data which doesn't follow middle hour convention? You already have to do that check to shift the last hour. Something like this should do it:

```py
# collect hourly data
for line in epwin:
    data = line.strip().split(',')
    year, month, day, hour = map(int, data[:4])

    # in an epw file year can be different for each month
    # since I'm using this timestamp as the key and will be using it for
    # sorting. I'm setting it up to 2015 - the real year will be collected
    # under modelYear
    timestamp = DateTime(month, day, hour - 1)

    for field_number in xrange(self._num_of_fields):

        value_type = EPWFields.field_by_number(field_number).value_type

        try:
            value = value_type(data[field_number])
        except ValueError as e:
            # failed to convert the value for the specific TypeError
            if value_type != int:
                raise ValueError(e)
            value = int(round(float(data[field_number])))

        self._data[field_number].append(DataPoint(value, timestamp_hour))

# move last item to start position for fields on the hour
for field_number in xrange(self._num_of_fields):
    middle_hour = EPWFields.field_by_number(field_number).middle_hour
    if middle_hour is False:
        # shift datetimes for an hour
        for data in self._data[field_number]:
            try:
                data.datetime = data.datetime.add_hour(1)
            except ValueError:
                # this is the last hour
                data.datetime = DateTime(1, 1, 0)

        # now move the last hour to first
        last_hour = self._data[field_number].pop()
        self._data[field_number].insert(0, last_hour)
```

Reply:
Ok.  I can see how this is just as efficient as the implementation I had before but it's easier to read.  I've changed it to be just like this.